src/chunk.c

```diff
@@ -4613,3 +4618,70 @@ ts_chunk_attach_osm_table_chunk(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(ret);
 }
+
+static ScanTupleResult
+chunk_tuple_osm_chunk_found(TupleInfo *ti, void *arg)
+{
+	bool isnull;
+
+	Datum osm_status_datum = slot_getattr(ti->slot, Anum_chunk_status, &isnull);
+	Assert(!isnull);
+	int32 osm_status = DatumGetInt32(osm_status_datum);
+	bool is_dropped = DatumGetBool(slot_getattr(ti->slot, Anum_chunk_dropped, &isnull));
+	if (is_dropped || !ts_flags_are_set_32(osm_status, CHUNK_STATUS_FOREIGN))
+		return SCAN_CONTINUE;
+
+	int *chunk_id = (int *) arg;
+	Datum chunk_id_datum = slot_getattr(ti->slot, Anum_chunk_id, &isnull);
+	Assert(!isnull);
+	*chunk_id = DatumGetInt32(chunk_id_datum);
+
+	return SCAN_DONE;
+}
+
+/* get OSM chunk id associated with the hypertable */
+int
+ts_chunk_get_osm_chunk_id(int hypertable_id)
+{
+	int chunk_id = INVALID_CHUNK_ID;
+	ScanKeyData scankey[2];
+	int32 osm_chunk_status = CHUNK_STATUS_FOREIGN;
+	Catalog *catalog = ts_catalog_get();
+	int num_found;
+	ScannerCtx scanctx = {
+		.table = catalog_get_table_id(catalog, CHUNK),
+		.index = catalog_get_index(catalog, CHUNK, CHUNK_STATUS_HYPERTABLE_ID_INDEX),
+		.nkeys = 2,
+		.scankey = scankey,
+		.data = &chunk_id,
+		.filter = NULL,
+		.tuple_found = chunk_tuple_osm_chunk_found,
+		.lockmode = AccessShareLock,
+		.scandirection = ForwardScanDirection,
+	};
+
+	/*
+	 * Perform an index scan on status + hypertable ID
+	 */
+	ScanKeyInit(&scankey[0],
+				Anum_chunk_status_hypertable_id_idx_status,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(osm_chunk_status));
```

Comment:
Since chunk.status is a bit field this is not correct and might bite us in the future.

Reply:
The bit operator, biteq , works on VARBIT type - this will not work with the int status type.
(see: https://www.postgresql.org/docs/current/datatype-bit.html).

Looks like I have to go back to the previous approach of adding a new column instead of reusing the status bits to identify an OSM chunk

Reply:
Is it ever combined with other statuses? Maybe we can assert that it's not and use it this way. It's kinda hackish, but adding another column will inevitably bloat the catalog and slow down all queries. Not sure by how much, but chunk catalog lookups do constitute a significant part of planning time already.